analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/HardcodedBytesRuleBase.cs

```diff
@@ -92,10 +100,10 @@ protected ProgramState ProcessStringToBytes(ProgramState state, IInvocationOpera
         bool IsConvertFromBase64String() =>
             invocation.TargetMethod.Name == nameof(Convert.FromBase64String)
             && invocation.TargetMethod.ContainingType.Is(KnownType.System_Convert)
-            && invocation.ArgumentValue("s") is { ConstantValue.HasValue: true };
+            && ArgumentIsHardcoded("s");
 
         bool ArgumentIsHardcoded(string parameterName) =>
             invocation.ArgumentValue(parameterName) is { } value
-            && state[value]?.HasConstraint(Hardcoded) is true;
+            && (value.ConstantValue.HasValue || state[value]?.HasConstraint(Hardcoded) is true);
```

Comment:
Added support for `Convert`, similar to `Encoding.GetBytes`, and simplified the checks a bit.